### PR TITLE
Feature #1 rest

### LINKED ISSUE #1: REST operations

### PR DESCRIPTION
**Pull request by anttileppa**
Resolves #1

### CODE REVIEW

File: src/main/java/fi/metatavu/metamind/liquibase/LiquibaseProducer.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  @LiquibaseType
+  public CDILiquibaseConfig createConfig() {
+    CDILiquibaseConfig config = new CDILiquibaseConfig();
+    config.setChangeLog("fi/metatavu/metamind/changelog.xml");
+    return config;
+  }
+  
+  @Produces
+  @LiquibaseType
+  public DataSource createDataSource() throws SQLException {
```
Comment (sonarq) [inline src/main/java/fi/metatavu/metamind/liquibase/LiquibaseProducer.java]: MINOR Remove the declaration of thrown exception 'java.sql.SQLException', as it cannot be thrown from method's body. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

**Comment by sonarq** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MetamindBotConfiguration.java#L80](https://github.com/Metatavu/metamind-api/blob/0036dc6539084c6c5101e9355adf36bb5bab5335/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L80): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MINOR][MINOR] [MetamindBotConfiguration.java#L96](https://github.com/Metatavu/metamind-api/blob/0036dc6539084c6c5101e9355adf36bb5bab5335/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L96): Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)
1. ![MINOR][MINOR] [MetamindBotConfiguration.java#L96](https://github.com/Metatavu/metamind-api/blob/0036dc6539084c6c5101e9355adf36bb5bab5335/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L96): Immediately return this expression instead of assigning it to the temporary variable "globalIntents". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)
1. ![INFO][INFO] [MetamindBotConfiguration.java#L103](https://github.com/Metatavu/metamind-api/blob/0036dc6539084c6c5101e9355adf36bb5bab5335/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L103): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by sonarq** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MetamindBotConfiguration.java#L80](https://github.com/Metatavu/metamind-api/blob/49e0fa222e173c02a76a979105fffd47852dbe4d/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L80): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MINOR][MINOR] [MetamindBotConfiguration.java#L96](https://github.com/Metatavu/metamind-api/blob/49e0fa222e173c02a76a979105fffd47852dbe4d/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L96): Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)
1. ![MINOR][MINOR] [MetamindBotConfiguration.java#L96](https://github.com/Metatavu/metamind-api/blob/49e0fa222e173c02a76a979105fffd47852dbe4d/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L96): Immediately return this expression instead of assigning it to the temporary variable "globalIntents". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)
1. ![INFO][INFO] [MetamindBotConfiguration.java#L103](https://github.com/Metatavu/metamind-api/blob/49e0fa222e173c02a76a979105fffd47852dbe4d/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L103): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/fi/metatavu/metamind/persistence/dao/MessageDAO.java [hunk 2]
```diff
@@ -14,13 +14,42 @@
...
     message.setExternalId(externalId);
-    message.setHint(hint);
     message.setSession(session);
     return persist(message);
   }
-  
+
+  /**
+   * Updates response
+   *
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/persistence/dao/MessageDAO.java]: Message param missing

File: src/main/java/fi/metatavu/metamind/persistence/dao/MessageDAO.java [hunk 3]
```diff
@@ -14,13 +14,42 @@
...
+   * @return updated message
+   */
+  public Message updateResponse(Message message, String response) {
+    message.setResponse(response);
+    return persist(message);
+  }
+
+  /**
+   * Updates hint
+   *
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/persistence/dao/MessageDAO.java]: Message param missing

**Comment by sonarq** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [MetamindBotConfiguration.java#L80](https://github.com/Metatavu/metamind-api/blob/ccd18ed4d64aa0a4dcfeaa393abe6b14d9510af4/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L80): Define and throw a dedicated exception instead of using a generic one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)
1. ![MINOR][MINOR] [MetamindBotConfiguration.java#L96](https://github.com/Metatavu/metamind-api/blob/ccd18ed4d64aa0a4dcfeaa393abe6b14d9510af4/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L96): Replace the type specification in this constructor call with the diamond operator ("<>"). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)
1. ![MINOR][MINOR] [MetamindBotConfiguration.java#L96](https://github.com/Metatavu/metamind-api/blob/ccd18ed4d64aa0a4dcfeaa393abe6b14d9510af4/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L96): Immediately return this expression instead of assigning it to the temporary variable "globalIntents". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)
1. ![INFO][INFO] [MetamindBotConfiguration.java#L103](https://github.com/Metatavu/metamind-api/blob/ccd18ed4d64aa0a4dcfeaa393abe6b14d9510af4/src/main/java/fi/metatavu/metamind/bot/MetamindBotConfiguration.java#L103): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**